.github/workflows/docker-build-push-multiarch.yml

```diff
@@ -0,0 +1,344 @@
+        value: ${{ jobs.build-and-push.outputs.tags }}
+      version:
+        description: "Generated Docker image version (from docker/metadata-action)"
+        value: ${{ jobs.build-and-push.outputs.version }}
+      runner_arches:
+        description: "The list of OS used to build images (for mapping to self hosted runners)"
+        value: ${{ jobs.prepare-matrix.outputs.runner_arches }}
+
+jobs:
+  prepare-matrix:
+    runs-on: ubuntu-arm64-small
+    outputs:
+      runner_arches: ${{ steps.matrix.outputs.runner_arches }}
+    steps:
+      - id: matrix
+        run: |
+          # This converts the potential list of `linux/arm64,linux/amd64` into a matrix of `arm64,x64`
+          # to match our hosted runner labels
+          MATRIX=$(echo "$PLATFORMS" | tr -d ' ' | tr ',' '\n' \
+            | jq -R 'if .=="linux/amd64" then "x64" elif .=="linux/arm64" then "arm64" else . end' \
+            | jq -s -c .)
+
+          # Set output
+          echo "runner_arches=$MATRIX" | tee -a "${GITHUB_OUTPUT}"
+        env:
+          PLATFORMS: ${{ inputs.platforms }}
+
+  build-and-push:
+    needs: prepare-matrix
+    strategy:
+      fail-fast: false
+      matrix:
+        arch: ${{ fromJson(needs.prepare-matrix.outputs.runner_arches) }}
+    runs-on: ubuntu-${{ matrix.arch }}-${{ inputs.server-size }}
+    env:
+      ARCH_TO_PLATFORM_MAP: '{"arm64": "linux/arm64", "x64": "linux/amd64"}' # TODO: Document this somewhere
+    outputs:
+      annotations: ${{ steps.build.outputs.annotations }}
+      digest: ${{ steps.build.outputs.digest }}
+      imageid: ${{ steps.build.outputs.imageid }}
+      images: ${{ steps.build.outputs.images }}
+      json: ${{ steps.build.outputs.json }}
+      labels: ${{ steps.build.outputs.labels }}
+      metadata: ${{ steps.build.outputs.metadata }}
+      metadatajson: ${{ steps.build.outputs.metadatajson }}
+      tags: ${{ steps.build.outputs.tags }}
+      version: ${{ steps.build.outputs.version }}
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - name: Prepare
+        run: |
+          PLATFORM=$(echo "$ARCH_TO_PLATFORM_MAP" | jq -r --arg arch "${MATRIX_ARCH}" '.[$arch]')
+          echo "PLATFORM=${PLATFORM}" | tee -a "${GITHUB_ENV}"
+          echo "PLATFORM_PAIR=${PLATFORM//\//-}" | tee -a "${GITHUB_ENV}"
+        env:
+          MATRIX_ARCH: ${{ matrix.arch }}
+      - name: Checkout code
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          persist-credentials: false
+      - name: Run custom build script if it exists
+        if: ${{ inputs.pre-build-script != '' }}
+        run: |
+          set -euo pipefail
+          echo "Running $PRE_BUILD_SCRIPT_PATH"
+
+          if [ -f "$PRE_BUILD_SCRIPT_PATH" ]; then
+            chmod +x "$PRE_BUILD_SCRIPT_PATH"
+            "$PRE_BUILD_SCRIPT_PATH"
+          else
+            echo "No script found at $PRE_BUILD_SCRIPT_PATH, skipping."
```

Comment:
Is skipping enough here or should this actually be an error? It does hint a mistake in the configuration.